syncserver/src/server/mod.rs

```diff
@@ -371,3 +396,92 @@ pub fn build_cors(settings: &Settings) -> Cors {
 
     cors
 }
+
+/// Emit database pool and threadpool metrics periodically
+fn spawn_metric_periodic_reporter<T: GetPoolState + Send + 'static>(
+    interval: Duration,
+    metrics: StatsdClient,
+    pool: T,
+    blocking_threadpool: Arc<BlockingThreadpool>,
+) -> Result<(), DbError> {
+    let hostname = hostname::get()
+        .expect("Couldn't get hostname")
+        .into_string()
+        .expect("Couldn't get hostname");
+    let blocking_threadpool_size =
+        str::parse::<u64>(&env::var("ACTIX_THREADPOOL").unwrap()).unwrap();
+    tokio::spawn(async move {
+        loop {
+            let PoolState {
+                connections,
+                idle_connections,
+            } = pool.state();
+            metrics
+                .gauge_with_tags(
+                    "storage.pool.connections.active",
+                    (connections - idle_connections) as u64,
+                )
+                .with_tag("hostname", &hostname)
+                .send();
+            metrics
+                .gauge_with_tags("storage.pool.connections.idle", idle_connections as u64)
+                .with_tag("hostname", &hostname)
+                .send();
+
+            let active_threads = blocking_threadpool.active_threads();
+            let idle_threads = blocking_threadpool_size - active_threads;
+            metrics
+                .gauge_with_tags("blocking_threadpool.active", active_threads)
+                .with_tag("hostname", &hostname)
+                .send();
+            metrics
+                .gauge_with_tags("blocking_threadpool.idle", idle_threads)
+                .with_tag("hostname", &hostname)
+                .send();
+
+            time::delay_for(interval).await;
+        }
+    });
+
+    Ok(())
+}
+
+/// A threadpool on which callers can spawn non-CPU-bound tasks that block their thread (this is
+/// mostly useful for running I/O tasks). `BlockingThreadpool` intentionally does not implement
+/// `Clone`: `Arc`s are not used internally, so a `BlockingThreadpool` should be instantiated once
```

Comment:
An alternative would have been to encapsulate an `Arc<AtomicU64>` and allow clients to `clone` `BlockingThreadpool`s, but I thought it would be more obvious to have a single instance of `BlockingThreadpool` represent the threadpool in its entirety